recipe/meta.yaml

```diff
@@ -27,6 +27,10 @@ requirements:
     - {{ compiler('cxx') }}
     - file  # [arm64]
   host:
+    - azure-core-cpp
+    - azure-identity-cpp
+    - azure-storage-common-cpp
```

Comment:
I saw in a commit that you added this and `azure-core-cpp` because of build warnings. Does that indicate a problem with these packages?

Reply:
> I saw in a commit that you added this and azure-core-cpp because of build warnings.

Right. Here are the [build logs](https://dev.azure.com/jdblischak/feedstock-builds/_build/results?buildId=444&view=results) from the run on my fork that motivated me to explicitly add `azure-core-cpp` and `azure-storage-common-cpp` to the host requirements. The warnings look like the following:

```
   INFO (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libazure-storage-blobs.so found in conda-forge::azure-storage-blobs-cpp-12.7.0-h00ab1b0_0
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libazure-storage-common.so found in ['azure-storage-common-cpp']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['azure-storage-common-cpp'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libazure-core.so found in ['azure-core-cpp']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['azure-core-cpp'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
   INFO (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libxml2.so.2 found in conda-forge::libxml2-2.11.5-h0d562d8_0
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libcurl.so.4 found in ['libcurl']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libcurl'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
```

> Does that indicate a problem with these packages?

I honestly don't know. I first noticed these types of warnings in the staged-recipes PR. I assumed I could simply list the top-level dependencies in the recipe (eg not have to repeat `azure-core-cpp` in each SDK library recipe), but then I got these conda-build warnings. I [asked](https://github.com/conda-forge/staged-recipes/pull/23297#issuecomment-1631042659) how to handle the warnings, and the [response](https://github.com/conda-forge/staged-recipes/pull/23297#issuecomment-1631061594) was to simply add them to the host requirements.

In summary, it's unclear to me whether I need to actually explicitly include these recursive dependencies. It could just be conda-build being overly cautious, but it could also be that listing them is required for the `run_exports` to work properly.

Reply:
https://github.com/conda-forge/aws-sdk-cpp-feedstock/pull/800#issuecomment-1722569651 explains why we need them to be host dependencies.

Reply:
Great, thanks for following up!